presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java

```diff
@@ -163,6 +173,55 @@ protected String getTableSchemaName(ResultSet resultSet)
         return resultSet.getString("TABLE_CAT");
     }
 
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session));
+                ResultSet resultSet = getColumns(tableHandle, connection.getMetaData())) {
+            int allColumns = 0;
+            List<JdbcColumnHandle> columns = new ArrayList<>();
+            while (resultSet.next()) {
+                allColumns++;
+                String columnName = resultSet.getString("COLUMN_NAME");
+                JdbcTypeHandle typeHandle = new JdbcTypeHandle(
+                        resultSet.getInt("DATA_TYPE"),
+                        Optional.ofNullable(resultSet.getString("TYPE_NAME")),
+                        resultSet.getInt("COLUMN_SIZE"),
+                        resultSet.getInt("DECIMAL_DIGITS"),
+                        Optional.empty());
+                Optional<ColumnMapping> columnMapping = toPrestoType(session, connection, typeHandle);
+                log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", tableHandle.getSchemaTableName(), columnName, typeHandle, columnMapping);
+                // skip unsupported column types
+                boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
+                // Note: some databases (e.g. SQL Server) do not return column remarks/comment here.
+                Optional<String> comment = Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
+                if (!comment.isPresent() && typeHandle.getJdbcType() == Types.DECIMAL && getDecimalRounding(session) == ALLOW_OVERFLOW && typeHandle.getColumnSize() > Decimals.MAX_PRECISION) {
+                    comment = Optional.of(format("source type is decimal(%s,%s)", typeHandle.getColumnSize(), typeHandle.getDecimalDigits()));
```

Comment:
comment has a strict semantic. Users may use it in production workflows, hence here some custom values is not a good idea. I would prefer to use column properties for that. That case you could have a dedicated for column property for data type from underlying data source in jdbc connector. That could work not only for decimal. Notice that this could be helpful whenever `CONVERT_TO_VARCHAR` is used. Also I don't think we should block this PR on this change, I would extract this as separate pull requests.

Reply:
Adding a new column properties with specific property for the underlying data_type is a good idea. I will remove the last commit that using the comment column and leave the implementation for a separate PR